### PR TITLE
regex tag scanner: Add initial test for detection/tagging of json responses

### PR DESCRIPTION
**Pull request by kingthorin**
Related to zaproxy/zaproxy#8399

https://github.com/zaproxy/zaproxy/blob/02b5c4f031e8cd6a741fa023310b68dd47d4dc7d/zap/src/main/resources/org/zaproxy/zap/resources/config.xml#L116-L125

### CODE REVIEW

File: zap/src/test/java/org/zaproxy/zap/extension/pscan/scanner/RegexAutoTagScannerUnitTest.java [hunk 1]
```diff
@@ -110,4 +135,43 @@ void shouldNotCountWhenDisabledThoughBodyContainsMatch()
...
+                return tagRule;
+            }
+        }
+        return null;
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @ValueSource(
+            strings = {
+                "application/json",
```
Comment (kingthorin) [inline zap/src/test/java/org/zaproxy/zap/extension/pscan/scanner/RegexAutoTagScannerUnitTest.java:151]:
```suggestion
                "application/JSON",
```

**Comment by psiinon** [comment]
This will only apply to new installs right?
We will need to update existing config files as part of the upgrade?

**Comment by kingthorin** [comment]
If the pattern is changed we would have to discuss doing that. For now this is just to get a base for testing.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**